### PR TITLE
fix: spring patch to version 5.3.18

### PR DESCRIPTION
**Pull request by witchpou**
ab#204

## Description

Patch for security leak, see https://www.heise.de/news/Verwirrung-um-kritische-Sicherheitsluecke-im-Umfeld-des-Spring-Framework-6658510.html

## Types of changes

- Security Patch

### CODE REVIEW

File: application/pom.xml [hunk 1]
```diff
@@ -94,7 +93,7 @@
             <plugin>
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
-                <version>${spring-version}</version>
+                <version>${spring-boot-version}</version>
```
Comment (witchpou) [inline application/pom.xml:96]: important change here

File: pom.xml [hunk 2]
```diff
@@ -1,210 +1,215 @@
...
-    <spring-version>2.3.12.RELEASE</spring-version>
-    <log4j2.version>2.17.0</log4j2.version>
-    <openapi-version>1.5.2</openapi-version>
-    <jackson-2-version>2.7.4</jackson-2-version>
-  </properties>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <spring-boot-version>2.3.12.RELEASE</spring-boot-version>
+        <spring-version>5.3.18</spring-version>
```
Comment (witchpou) [inline pom.xml:20]: change here

File: pom.xml [hunk 3]
```diff
@@ -1,210 +1,215 @@
...
+        <log4j2.version>2.17.2</log4j2.version>
```
Comment (witchpou) [inline pom.xml:21]: change from 2.17.0 to 2.17.2

File: pom.xml [hunk 4]
```diff
@@ -1,210 +1,215 @@
...
-        <version>7.15.0</version>
-      </dependency>
-      <dependency>
-        <groupId>org.freemarker</groupId>
-        <artifactId>freemarker</artifactId>
-        <version>2.3.31</version>
-      </dependency>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
```
Comment (witchpou) [inline pom.xml:37]: added Spring-core dependency separately

**Review by DerCed — APPROVED** [review]

**Pull request merged**